airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/MergeUnions.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.data
+
+class MergeUnions : AirbyteSchemaIdentityMapper {
+    override fun mapUnion(schema: UnionType): AirbyteType {
+        // Map the options first so they're in their final form
+        val mappedOptions = schema.options.map { map(it) }
+
+        // Add all the mapped options to a new set, merging
+        val newOptions = mutableSetOf<AirbyteType>()
+        mergeOptions(newOptions, mappedOptions)
+        if (newOptions.size == 1) {
+            return newOptions.first()
+        }
+        return UnionType(newOptions.toList())
+    }
+
+    private fun mergeOptions(into: MutableSet<AirbyteType>, from: List<AirbyteType>) {
+        for (option in from) {
+            if (option is UnionType) {
+                // If this is a union of a union, recursively merge the other union's options in
+                mergeOptions(into, option.options)
+            } else if (option is ObjectType) {
+                val objMaybe = into.find { it is ObjectType }
```

Comment:
nit: you could probably avoid the explicit cast below?
```kotlin
val objMaybe: ObjectType? = into.find
```

(also, `existingObjOption` is a nicer name)